Include/internal/pycore_object.h

```diff
@@ -652,27 +654,64 @@ _PyObject_DictOrValuesPointer(PyObject *obj)
 static inline int
 _PyDictOrValues_IsValues(PyDictOrValues dorv)
 {
-    return ((uintptr_t)dorv.values) & 1;
+    return ((uintptr_t)FT_ATOMIC_LOAD_PTR_RELAXED(dorv.values)) & 1;
```

Comment:
I'm a bit confused by this: the previous code did not have a pointer load. I think this is doing an atomic load from a new pointer to a local variable -- that doesn't look right to me.

The atomic load should be in the caller to this function. By the time we get to this function we have a value.